src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/prison/PrisonApiClient.kt

```diff
@@ -113,4 +119,15 @@ class PrisonApiClient(@Qualifier("prisonApiWebClient") private val webClient: We
       .retrieve()
       .bodyToMono(InmateDetail::class.java)
       .block() ?: throw RuntimeException()
+
+  /**
+   * The prison-api end-point expects requests to have role 'TRANSFER_PRISONER', scope 'write' and a (NOMIS) username.
+   */
+  fun transferIn(offenderNo: String, detail: TransferIn) =
+    webClient.put()
+      .uri("/api/offenders/$offenderNo/transfer-in")
+      .bodyValue(detail)
+      .retrieve()
+      .toBodilessEntity()
+      .block() ?: throw RuntimeException()
```

Comment:
this should really have a message and the method above should as well, maybe an `IllegalStateException` with a message saying there is no response?

Reply:
ok